.github/workflows/containers.yml

```diff
@@ -56,31 +72,78 @@ jobs:
+      - name: Push untagged images
         if: github.event_name != 'release'
-        run: docker push ${{ env.PIPELINE_MANAGER_IMAGE }} && docker push ${{ env.KAFKA_CONNECT_IMAGE }}
+        run: docker push ${{ env.KAFKA_CONNECT_IMAGE }}
 
       # Tagged pipeline-manager image
-      - name: Push (pipeline-manager)
-        uses: docker/build-push-action@v4
-        if: github.event_name == 'release'
+      - name: Push pipeline-manager
+        id: build_pipeline_manager
+        uses: docker/build-push-action@v6
         with:
           context: .
+          platforms: ${{matrix.os}}/${{matrix.arch}}
           file: deploy/Dockerfile
           push: true
-          tags: ${{ steps.meta_pipeline_manager.outputs.tags }}
+          # Only add tags for release images
+          tags: ${{ github.event_name == 'release' && steps.meta_pipeline_manager.outputs.tags }}
           labels: ${{ steps.meta_pipeline_manager.outputs.labels }}
+          annotations: ${{ steps.meta_pipeline_manager.outputs.annotations }}
+
+      - name: Export multiarch images digests
+        id: export_image_digest
+        run: |
+          echo "digest-${{matrix.os}}-${{matrix.arch}}=${{ steps.build_pipeline_manager.outputs.digest }}" >> $GITHUB_OUTPUT
 
       - uses: actions/delete-package-versions@v4
         with:
           package-name: pipeline-manager
           package-type: 'container'
           min-versions-to-keep: 5
           delete-only-untagged-versions: 'true'
+
+    outputs:
+      digest-linux-amd64: ${{ steps.export_image_digest.outputs.digest-linux-amd64 }}
+      digest-linux-arm64: ${{ steps.export_image_digest.outputs.digest-linux-arm64 }}
+
+  push-docker-manifest:
+    needs: build-and-push-image
+    runs-on: blacksmith-2vcpu-ubuntu-2204-arm
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Docker meta
+        id: meta_pipeline_manager
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ env.PIPELINE_MANAGER_IMAGE }}
+          tags: |
+            type=semver,pattern={{version}}
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Push latest manifest
+        run: |
+            docker manifest create ${{ env.PIPELINE_MANAGER_IMAGE }} \
+              ${{ env.PIPELINE_MANAGER_IMAGE }}@${{needs.build-and-push-image.outputs.digest-linux-amd64}} \
+              ${{ env.PIPELINE_MANAGER_IMAGE }}@${{needs.build-and-push-image.outputs.digest-linux-arm64}}
+            docker manifest push ${{ env.PIPELINE_MANAGER_IMAGE }}:latest
+      - name: Push versioned manifest
+        if: github.event_name == 'release'
+        run: |
+            docker manifest create ${{ env.PIPELINE_MANAGER_IMAGE }} \
+              ${{ env.PIPELINE_MANAGER_IMAGE }}:linux-amd64 \
```

Comment:
I'm pretty sure this will always overwrite the same tags (pipeline-manager:linux-amd64 and pipeline-manager:linux-arm64?). I think what you want instead is pipeline-manager-amd64 and pipeline-manager-arm64, which will each have their own "latest" and tags.